src/main/java/com/forest/mytopmovies/utils/HttpUtil.java

```diff
@@ -26,8 +26,10 @@ public static String get(String baseUrl, String uri) throws TMDBHttpRequestExcep
         return bodySpec.exchangeToMono(clientResponse -> {
             if (clientResponse.statusCode().equals(HttpStatus.OK)) {
                 return clientResponse.bodyToMono(String.class);
+            } else if (clientResponse.statusCode().equals(HttpStatus.NOT_FOUND)) {
+                return clientResponse.createException().flatMap(error -> errorMono(error, "Movie is not found. Please double check with search api \"/api/v1/search/movies\""));
             } else {
-                return clientResponse.createException().flatMap(error -> Mono.error(() -> new TMDBHttpRequestException(error.getMessage())));
+                return clientResponse.createException().flatMap(error -> errorMono(error, "Failure when execute request to TMDB API"));
```

Comment:
"Failure when execute request to TMDB API" can be moved into errorMono method, then `flatMap(error -> errorMono, "...")` can become `flatMap(this::errorMono)` or something similar.